MapboxStatic/Overlay.swift

```diff
@@ -201,31 +201,31 @@ public class Marker: MarkerImage, Point {
  The marker image is always centered on the specified location. When creating an asymmetric marker like a pin, make sure that the tip of the pin is at the center of the image.
  */
 @objc(MBCustomMarker)
-public class CustomMarker: NSObject, Overlay {
+open class CustomMarker: NSObject, Overlay {
     /// The geographic coordinate to place the marker at.
-    public var coordinate: CLLocationCoordinate2D
+    open var coordinate: CLLocationCoordinate2D
     
     /**
      The HTTP or HTTPS URL of the image.
      
      The API caches custom marker images according to the `Expires` and `Cache-Control` headers. If you host the image on your own server, make sure that at least one of these headers is set to an proper value to prevent repeated requests for the image.
      */
-    public var URL: NSURL
+    open var URL: Foundation.URL
     
     /**
      Initializes a marker with the given coordinate and image URL.
      
      - parameter coordinate: The geographic coordinate to place the marker at.
      - parameter URL: The HTTP or HTTPS URL of the image.
      */
-    public init(coordinate: CLLocationCoordinate2D, URL: NSURL) {
+    public init(coordinate: CLLocationCoordinate2D, URL: Foundation.URL) {
         self.coordinate = coordinate
         self.URL = URL
     }
     
-    public override var description: String {
-        #if swift(>=2.3)
-            let escapedURL = URL.absoluteString!.stringByAddingPercentEncodingWithAllowedCharacters(allowedCharacterSet)!
+    open override var description: String {
+        #if swift(>=3.0)
+            let escapedURL = URL.absoluteString.addingPercentEncoding( withAllowedCharacters: .urlHostAllowed)!
         #else
```

Comment:
Remove this Swift 2.2 case, since this PR forces client code to use Swift 3 at a minimum.